pkg/client/client.go

```diff
@@ -187,7 +203,20 @@ func getExecCredentialWithToken(token, expirationTimestamp string) (*datamodel.E
 	}, nil
 }
 
-func loadExecCredential() (*datamodel.ExecCredential, error) {
+type ClientHelpers interface {
+	LoadExecCredential() (*datamodel.ExecCredential, error)
+	GetServerURL(execCredential *datamodel.ExecCredential) (string, error)
+	GetTLSConfig(pemCAs []byte, nextProto string, insecureSkipVerify bool) (*tls.Config, error)
+	LoadAzureJSON() (*datamodel.AzureConfig, error)
+}
+
+type clientHelpersImpl struct{}
+
+func NewClientHelpers() ClientHelpers {
+	return &clientHelpersImpl{}
+}
+
+func (c *clientHelpersImpl) LoadExecCredential() (*datamodel.ExecCredential, error) {
```

Comment:
I'm pretty sure we can mock the environment variable set: https://stackoverflow.com/questions/66061175/mocking-os-getenvenv-var